src/pass/loop_partition.cc

```diff
@@ -324,6 +375,98 @@ class LoopPartitioner : public IRMutator {
   CandidateSelector selector;
 };
 
+// Returns an interval (in the first component) in which all the conditions
+// given in the second component provably have value given by cond_value
+std::pair<IntSet, std::unordered_set<const Node*>>
+LoopPartitioner::GetIntervalAndCondset(const Partition &partitions,
+                                       const arith::Interval &for_interval,
+                                       bool cond_value) {
+  Array<IntSet> sets;
+  std::unordered_set<const Node*> cond_set;
+
+  for (const auto &kv : partitions) {
+    if (kv.first.second == cond_value) {
+      arith::Interval interval = kv.second.as<arith::IntervalSet>()->i;
+      auto intersection = arith::Interval::make_intersection(interval, for_interval);
+
+      // TODO(derisavi): the following if statement needs to be removed as soon as
```

Comment:
Not yet. The commit of HalideIR currently used in master branch of tvm is 55ba177 which is older than a768f2f0 that is required to remove that if statement.

Reply:
can you update the submodule then remove here?

Reply:
I tried to pick latest commit in HalideIR but apparently because of this commit (https://github.com/dmlc/HalideIR/commit/92ef20fdb617e6a2e530131084cf665f810eca65) , TVM source code compilation gives an error. I once tried to fix this compile problem but it wasn't trivial and required understanding the code change in HalideIR. (@jroesch can you please take a look and help us out?)

So we can either 1) wait for this compile problem to be fixed, perform the TODO, and then merge this PR or 2) merge this PR, wait for the compile problem to be fixed and then perform the TODO? I'm open to both but I prefer (2).

Reply:
@derisavi Okay, let us merge the PR for now

Reply:
@derisavi Could you try other commit  like `HEAD~1` besides the head

Reply:
The change I need it at `HEAD` and the problematic commit is `HEAD~1` and TVM is currently using `HEAD~2`. So trying `HEAD~1` wouldn't resolve the problem.

Reply:
I rebased/squashed and pushed with the TODO item still in the code.